lib/index.ts

```diff
@@ -40,12 +50,21 @@ async function buildDepTree(targetFileRaw: string, lockFileRaw: string): Promise
     depTree.dependencies[dep] = await buildSubTreeRecursive(dep, [], lockFile);
   }));
 
+  if (includeDev && targetFile.devDependencies) {
+    const topLevelDevDeps = Object.keys(targetFile.devDependencies);
+    depTree.hasDevDependencies = true;
+    await Promise.all(topLevelDevDeps.map(async (dep) => {
```

Comment:
relevant also for the prod dependencies - why do we need the `Promise.all()` and `await` here? it will all still run on a single thread and doesn't do any async IO. So it only creates unnecessary overhead I think

Reply:
I was following @darscan suggestion `do it async from the beginning, because it's much harder to do it later` and I agree with that - overhead now can save us a toil in the future, which is too far away to say, whether I/O operations will be introduced or not.